lib/modules/manager/github-actions/extract.ts

```diff
@@ -7,7 +7,7 @@ import type { PackageDependency, PackageFile } from '../types';
 
 const dockerRe = regEx(/^\s+uses: docker:\/\/([^"]+)\s*$/);
 const actionRe = regEx(
-  /^\s+-?\s+?uses: (?<replaceString>['"]?(?<depName>[\w-]+\/[\w-]+)(?<path>\/.*)?@(?<currentValue>[^\s'"]+)['"]?(?:\s+#\s+(?:renovate:\s+)?tag=(?<tag>\S+))?)/
+  /^\s+-?\s+?uses: ['"]?(?<replaceString>(?<depName>[\w-]+\/[\w-]+)(?<path>\/.*)?@(?<currentValue>[^\s'"]+)(?:['"]?\s+#\s+(?:renovate:\s+)?tag=(?<tag>\S+))?)/
```

Comment:
i think this will cause issues with our tagging comment. did you verified the pinned dep scenario in a real repo?

Reply:
You mean like https://github.com/renovatebot/renovate/blob/ab8a9156007707f5cbe061515ab847e7f290c3bc/.github/workflows/build.yml#L66 ?

Reply:
yes, or

```yaml
uses: 'actions/setup-node@56337c425554a6be30cdef71bf441f15be286854' # tag=v3.1.1 
uses: "actions/setup-node@56337c425554a6be30cdef71bf441f15be286854" # tag=v3.1.1 
``

Reply:
Would it be better if I convert this to `js-yaml` parsing instead of line by line regex like we did for gitlab-ci?
I thought to use the `js-yaml` library to read the file and also do some preprocessing on the file to fill a data structure that maps `uses` deps to comments if there are some. 
wdyt @viceice @rarkins ?

Reply:
I don't think this makes it easier, as we need to parse those renovate tag comments to get the current version